### PR TITLE
Gnadtech 2349 2

### PR DESCRIPTION
**Pull request by TimHewitt**
Resolves GNADTECH-2349

**Overall change:**
Add Neilsen analytics to story/article AMP pages in Australia

**Code changes:**

Created a NielsenAnalytics component that writes some JSON to the page if on an AMP page  and nielsenAnalytics toggle is enabled.

**Testing:**

- Check that values are only sent to Neilsen on AMP pages when in Australia
- Check correct values are sent (see jira ticket above)

The below will result in calls to URLs containing **imrworldwide** along with other bits of metadata added by Neilsen themselves....

----

**Comments from Stephen Macleod:**

Geo limited to Australia
Pages limited to text (so do not need to fire on video events on embedded players)
If we can, set the reserved variables below with the same values as on web pages
 
**Required Variables and Values**

APID  (note this will also serve as the production ID once validated by Nielsen) 

News 474C2B0B-0C04-4182-BCFB-FC9469A48C9B
Sport DC4AFDB2-B352-4D51-81EF-38BE41114F22
 

Below are examples of browser static metadata objects that would need to be created for each page and called along with the 'staticstart' function in order to generate a page view -

https://www.google.co.uk/amp/s/www.bbc.com/news/business-58007120.amp

var staticMetadataObject = { 
  type: "static", //Required: the same for every static asset 
  section: "BBC News Business", //Required
  segC: "BBC – Google AMP"
};

Static metadata example for - https://www.bbc.com/sport/formula1/51607740 

var staticMetadataObject = { 
  type: "static", //Required: the same for every static asset 
  section: "BBC Sport Formula 1", //Required
  segC: "BBC – Google AMP"
};

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [x] This PR requires manual testing (to be done by BBC Studios/GNL)

### CODE REVIEW

File: src/app/lib/config/toggles/liveConfig.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ export default {
   navOnArticles: {
     enabled: true,
   },
+  nielsenAnalytics: {
```
Comment (jroebu14) [inline src/app/lib/config/toggles/liveConfig.js:32]: I'll need to add this new toggle in our toggles UI

File: src/app/containers/NielsenAnalytics/Amp/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { render } from '@testing-library/react';
+import AmpNielsenAnalytics from '.';
+
+describe('Assertions', () => {
+  it('should render Neilsen amp-analytics component', () => {
+    render(
+      <AmpNielsenAnalytics apid="apid" assetid="assetid" section="section" />,
+    );
+
+    const ampNeilsenEl = document.querySelector('amp-analytics');
```
Comment (jroebu14) [inline src/app/containers/NielsenAnalytics/Amp/index.test.jsx:11]: mind if we update these query selectors to use a RTL query instead? I think probably this one https://testing-library.com/docs/queries/bytestid e.g.

```html
<div data-testid="custom-element" />
```

```jsx
import {render, screen} from '@testing-library/react'

render(<MyComponent />)
const element = screen.getByTestId('custom-element')
```

**Review by jroebu14 — APPROVED** [review]
Looks good. Thanks for this

File: src/app/containers/NielsenAnalytics/Amp/index.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { string } from 'prop-types';
+
+// styled-components removes non-standard attributes (such as AMP attributes) on
+// server rendering. spreading props like this allows us to add AMP attributes
+// to the element.
+const AccessDiv = props => <div {...props} />;
+
+// Nielsen should only run in AU,
+// Using amp-geo to hide/display based on the country
+// amp-analytics doesn't generate any call if display none
```
Comment (andrewscfc) [inline src/app/containers/NielsenAnalytics/Amp/index.jsx:12]: Be good to confirm this manually in testing, afaik the `display: none` does not display the element but leaves it in the DOM, the docs for `amp-geo` don't specifically call out `amp-analytics` for this behaviour https://amp.dev/documentation/components/amp-geo/
image

File: src/app/containers/NielsenAnalytics/Amp/index.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+const DisplayWrapper = styled(AccessDiv)`
+  .amp-geo-pending &,
+  .amp-geo-group-gbOrUnknown & {
+    display: none;
+    visibility: hidden;
+  }
+  .amp-iso-country-au & {
+    display: block;
+  }
```
Comment (andrewscfc) [inline src/app/containers/NielsenAnalytics/Amp/index.jsx:18]: So this covers:
- Don't display when loading
- Don't display in GB or unknown location
- Display in australia

What stops the analytics element appearing in every other country outside GB? It perhaps needs a default `display: none; visibility: hidden` before the more specific selector for AU?

File: src/app/lib/config/toggles/liveConfig.js (hunk 1 above)
Comment (TimHewitt) [reply src/app/lib/config/toggles/liveConfig.js:32]: OK thanks. I wondered if/where an external versions of the toggles were!

File: src/app/containers/NielsenAnalytics/Amp/index.test.jsx (hunk 2 above)
Comment (TimHewitt) [reply src/app/containers/NielsenAnalytics/Amp/index.test.jsx:11]: Go for it! I just copied the pattern from another test.

File: src/app/containers/NielsenAnalytics/index.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  if (lastSection.includes('-')) {
+    // CPS
+    const lastSectionSplit = lastSection.split('-');
+    section = `${section} ${capitalise(lastSectionSplit[0])}`;
+  }
+
+  return (
+    <AmpNielsenAnalytics apid={apid} assetid={assetid} section={section} />
+  );
+};
```
Comment (andrewscfc) [inline src/app/containers/NielsenAnalytics/index.jsx:56]: I think the logic in here should be explicitly tested missing coverage includes:
- Correct `apid` for news and sport services
- Removing `.amp` from the `assetid`
- Logic to determine the `section`, I can't really understand fully what it's intent is

Comment (TimHewitt) [reply src/app/containers/NielsenAnalytics/index.jsx:56]: I've added tests for the apid.
I'd rather leave the assetid ones for now - there are ongoing conversations with Nielsen on what these should be and why the currently aren't getting sent.

File: src/app/lib/config/toggles/liveConfig.js (hunk 1 above)
Comment (jroebu14) [reply src/app/lib/config/toggles/liveConfig.js:32]: @TimHewitt I know it's stated in ticket but is adding a toggle for this absolutely necessary? No worries if it is. Just don't want to add and maintain another toggle if we probably won't toggle this feature on/off

File: src/app/containers/NielsenAnalytics/Amp/index.jsx (hunk 3 above)
Comment (TimHewitt) [reply src/app/containers/NielsenAnalytics/Amp/index.jsx:12]: Tested

File: src/app/containers/NielsenAnalytics/Amp/index.jsx (hunk 4 above)
Comment (TimHewitt) [reply src/app/containers/NielsenAnalytics/Amp/index.jsx:18]: Fixed

File: src/app/containers/NielsenAnalytics/index.jsx (hunk 5 above)
Comment (andrewscfc) [reply src/app/containers/NielsenAnalytics/index.jsx:56]: Okay @TimHewitt , could you please revisit in a later PR to add them? Given this is an area we as a team have little knowledge of having tests to describe the requirements is even more important.

**Review by andrewscfc — APPROVED** [review]
LGTM but could you please consider @jroebu14's comment around toggles: https://github.com/bbc/simorgh/pull/9400#discussion_r705377061

File: src/app/lib/config/toggles/liveConfig.js (hunk 1 above)
Comment (TimHewitt) [reply src/app/lib/config/toggles/liveConfig.js:32]: Sorry @jroebu14 missed this...
I think it is required due to potential GDPR issues, but will double check...

Comment (TimHewitt) [reply src/app/lib/config/toggles/liveConfig.js:32]: From John Horth: 
_My understanding was that we should always have a means to readily disable third party integrations. In this instance where user data is being handled, and global legislation constantly evolving that req was amplified._

Comment (jroebu14) [reply src/app/lib/config/toggles/liveConfig.js:32]: I believe @aeroplanejane has now set this toggle up 👍

**Comment by aeroplanejane** [comment]
@jroebu14 does this need another review before the changes are committed?

**Review by ryanmccombe — APPROVED** [review]

**Comment by aeroplanejane** [comment]
Think this can be closed @TimHewitt ?

**Pull request merged**